### PR TITLE
Allow compiler executable under test to be overridden

### PR DESCRIPTION
**Pull request by HertzDevil**
This PR adds an environment variable `SPEC_COMPILER` which, when set, overrides the Crystal compiler executable being used in specs.

This is mainly relevant for local development on Windows. Unlike on CI, `bin\crystal.exe` relative to the repository root is normally not expected to exist, because the local compiler is supposed to go into `.build` instead; also unlike on Unix, `bin\crystal` must resolve to an `.exe`, never some other script like `bin\crystal.bat`. With this PR, any installation of Crystal or wrapper script can be used in place of `bin/crystal`.

`crystal spec` automatically sets this environment variable to the compiler process itself if the variable is absent, before building the specs (not just before running, so that `SPEC_COMPILER` would always be available even in macros). If the specs are built and then run separately, like on CI, then `SPEC_COMPILER` should also be set beforehand (our workflows don't need this).

### CODE REVIEW

**Comment by straight-shoota** [comment]
Sounds good! I think the env var name should be `CRYSTAL_SPEC_COMPILER`, though.

**Comment by HertzDevil** [comment]
`CRYSTAL_SPEC_COMPILER_FLAGS` and `CRYSTAL_SPEC_COMPILER_THREADS` are already used by the `bin/check-compiler-flag`, but these are only effective for the fresh compiler class required by the specs, not any existing Crystal executable, so there would be some potential confusion.

**Comment by HertzDevil** [comment]
`CRYSTAL_SPEC_COMPILER_EXECUTABLE` might be better.

**Comment by Sija** [comment]
Perhaps a bit shorter  `CRYSTAL_SPEC_COMPILER_BIN`?

**Review by straight-shoota — APPROVED** [review]

File: spec/debug/driver.cr [hunk 1]
```diff
@@ -13,7 +13,7 @@ input = ARGV[0]
 bin = File.join(tmp_build_dir, "debug_test_case")
 debugger_script = File.join(tmp_build_dir, "./debugger.script")
 
-`#{repo_base_dir}/bin/crystal build --debug #{input} -o #{bin}`
+Process.run(ENV["CRYSTAL_SPEC_COMPILER_BIN"]? || "#{repo_base_dir}/bin/crystal", ["build", "--debug", input, "-o", bin])
```
Comment (oprypin) [inline spec/debug/driver.cr:16]: This was already problematic before, so feel free to ignore, but we now have a good opportunity to check that this command actually succeeded.

**Review by oprypin — APPROVED** [review]

**Pull request merged**